### PR TITLE
Piece check improvements

### PR DESCRIPTION
**Pull request by valeriegc**
Improved piece checks to include and notice where the opponent pieces ares situated. When reviewing please note that the white pieces are only testable, bc currently opponent turn is by default white.

### CODE REVIEW

**Comment by AlexanderKolberg** [comment]
Screenshot 2023-06-25 at 17 18 38

Tried to run your `pnpm lint `command, got a couple of prettier issues

**Comment by AlexanderKolberg** [comment]
The` pnpm check` command returns a couple of errors
Screenshot 2023-06-25 at 17 22 29

File: src/functions/moveChecks/bishopCheck.ts [hunk 1]
```diff
@@ -8,45 +8,70 @@ import {
...
+	diagonalRowNW,
+	hasOwnPiece,
+	hasOpponentPiece
 } from '../../global';
+import type { Square } from '../../stores';
 
-export const bishopCheck = (targetSquare: number) => {
+export const bishopCheck = (targetSquare: number, board: Square[], turn: string) => {
 	const tempArray: number[] = [];
 	const startSquare = targetSquare;
```
Comment (AlexanderKolberg) [inline src/functions/moveChecks/bishopCheck.ts]: Why are you renaming this here? If you want the name to be startSquare, you can just name it that in the arguments

**Comment by valeriegc** [comment]
> Screenshot 2023-06-25 at 17 18 38
> Tried to run your `pnpm lint `command, got a couple of prettier issues



> Screenshot 2023-06-25 at 17 18 38
> Tried to run your `pnpm lint `command, got a couple of prettier issues

Have ran pnpm format, issues fixed.

File: src/functions/moveChecks/bishopCheck.ts (hunk 1 above)
Comment (valeriegc) [reply src/functions/moveChecks/bishopCheck.ts]: resolved

**Comment by valeriegc** [comment]
> The` pnpm check` command returns a couple of errors Screenshot 2023-06-25 at 17 22 29

fixed

**Pull request merged**